codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/generators/IntEnumGenerator.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.ruby.codegen.generators;
+
+import java.util.function.Consumer;
+import software.amazon.smithy.codegen.core.directed.GenerateIntEnumDirective;
+import software.amazon.smithy.model.shapes.IntEnumShape;
+import software.amazon.smithy.ruby.codegen.GenerationContext;
+import software.amazon.smithy.ruby.codegen.RubySettings;
+
+public final class IntEnumGenerator
+        implements Consumer<GenerateIntEnumDirective<GenerationContext, RubySettings>> {
+    @Override
+    public void accept(GenerateIntEnumDirective<GenerationContext, RubySettings> directive) {
+        var settings = directive.context().settings();
```

Comment:
It seems like there is a lot of duplication across these shape generators.  Maybe some mechanism to share some of this boilerplate?